aten/src/ATen/CPUGenerator.cpp

```diff
@@ -14,16 +14,16 @@ namespace detail {
  * getDefaultCPUGenerator gets the default generator for a particular
  * device.
  */
-CPUGenerator* getDefaultCPUGenerator() {
-  static CPUGenerator default_gen_cpu(getNonDeterministicRandom());
-  return &default_gen_cpu;
+Generator getDefaultCPUGenerator() {
+  static auto default_gen_cpu = createCPUGenerator(c10::detail::getNonDeterministicRandom());
+  return default_gen_cpu;
```

Comment:
Not really sure if you should be returning `Generator` here or `const Generator&` here. Latter might be better as this generator always stays live and there's no reason to force a refcount bump here.

Reply:
addressed